source/NoteRefFilter.cpp

```diff
@@ -0,0 +1,49 @@
+// Header file
+#include "NoteRefFilter.h"
+
+// System libraries
+#include <String.h>
+#include <Entry.h>
+#include <Node.h>
+#include <NodeInfo.h>
+
+// Constructor
+NoteRefFilter :: NoteRefFilter() {
+}
+
+// Filter hook method
+bool NoteRefFilter :: Filter(const entry_ref *ref,
+			BNode *node,
+			struct stat_beos *stat,
+			const char *mimeType) {
+
+	// Follow links
+	BEntry entry(ref,true);
+
+	// Return true if the file is a directory
+	if (entry.IsDirectory())
+		return true;
+
+	// Create a BNode and a BNodeInfo to get the MIME type
+	BNode n(&entry);
+	BNodeInfo ninfo(&n);
+
+	// Create a string to hold the MIME type
+	BString m;
+
+	// Lock the buffer's string to use with BNodeInfo::GetType
+	char *mbuf = m.LockBuffer(B_MIME_TYPE_LENGTH);
+
+	// Get MIME type
+	ninfo.GetType(mbuf);
+
+	// Unlock the string's buffer for use 
+	m.UnlockBuffer();
```

Comment:
I tried using it but it just kept guessing the MIME type as `application/octet-stream`. Not sure why it couldn't guess `application/takenotes` properly

Reply:
Have you checked the FileTypes preferences if the right MIME type is known to the system?

Reply:
I did.

Reply:
It should work. Can you show us the relevant code you have?

Reply:
Just realized it actually does work but only if the file name ends in `.tkn`. Also I just pushed that code, it should be available [here](https://github.com/fantoro/TakeNotes/blob/0ac52f4ca205141767740021ae47bdb3669bf66a/source/NoteRefFilter.cpp#L37).

Reply:
```
	BMimeType m;

	// Guess the MIME type and store it in m
	if(BMimeType::GuessMimeType(&fRef, &m))
		return false;

	// Return true if the MIME type matches
	if (m == "application/takenotes")
```
The first if-statement should be `if (... == B_OK)`. The effect is a reverse logic because B_OK is 0.

You should get a warning for the second if-statement because BMimeType is not compatible with `const char*`. Instead, do this: `if (strcmp(m.Type(), "...") == 0)`.

Reply:
Ah I forgot to add `(... == B_OK)`, I'll fix it right away.

Also `BMimeType` is compatible with `const char*` according to the [Haiku book](https://www.haiku-os.org/docs/api/classBMimeType.html#a303623e252ecf9932ae2c735d6774add)

Reply:
Ah, BMimeType has operator functions for `==`, so the 2nd if-statement works.

Reply:
Alright made a commit fixing the missing `(... != B_OK)`.

Reply:
By the way I can't check if there are any other problems because I don't have any access to a PC (other than my phone) right now.

Reply:
Alright just checked if everything works and it seems to be working fine. I'll resolve this suggestion because I'm assuming that everything else is fine and doesn't need any further work.